contracts/test/LightClient.t.sol

```diff
@@ -0,0 +1,450 @@
+// SPDX-License-Identifier: Unlicensed
+
+/* solhint-disable contract-name-camelcase, func-name-mixedcase, one-contract-per-file */
+
+pragma solidity ^0.8.0;
+
+// Libraries
+import "forge-std/Test.sol";
+import { IPlonkVerifier as V } from "../src/interfaces/IPlonkVerifier.sol";
+
+// Target contract
+import { LightClient as LC } from "../src/LightClient.sol";
+import { LightClientTest as LCTest } from "./mocks/LightClientTest.sol";
+import { BN254 } from "bn254/BN254.sol";
+
+/// @dev Common helpers for LightClient tests
+contract LightClientCommonTest is Test {
+    LCTest lc;
+    uint32 public constant BLOCKS_PER_EPOCH_TEST = 3;
+    LC.LightClientState public genesis;
+    // this constant should be consistent with `hotshot_contract::light_client.rs`
+    uint64 internal constant STAKE_TABLE_CAPACITY = 10;
+
+    /// @dev initialized ledger like genesis and system params
+    function init() public {
+        string[] memory cmds = new string[](4);
+        cmds[0] = "diff-test";
+        cmds[1] = "mock-genesis";
+        cmds[2] = vm.toString(BLOCKS_PER_EPOCH_TEST);
+        cmds[3] = vm.toString(STAKE_TABLE_CAPACITY / 2);
```

Comment:
Why do we use STAKE_TABLE_CAPACITY/2 and not STAKE_TABLE_CAPACITY?

Reply:
I want to fill the stake table half-full to start with, so that we have more room for adding new registration later, but now it seems we don't have that need. I don't mind changing it to fully occupied to start with. I think both ways are fine.

Reply:
Ok let us keep it as is.